orchagent/mirrororch.cpp

```diff
@@ -613,6 +671,83 @@ bool MirrorOrch::updateSession(const string& name, MirrorEntry& session)
     return ret;
 }
 
+bool MirrorOrch::setUnsetPortMirror(Port port,
+                                    bool ingress,
+                                    bool set,
+                                    sai_object_id_t sessionId)
+{
+    sai_status_t status;
+    sai_attribute_t port_attr;
+    port_attr.id = ingress ? SAI_PORT_ATTR_INGRESS_MIRROR_SESSION:
+                           SAI_PORT_ATTR_EGRESS_MIRROR_SESSION;
+    if (set)
+    {
+        port_attr.value.objlist.count = 1;
+        port_attr.value.objlist.list = (long unsigned int *)calloc(port_attr.value.objlist.count, sizeof(sai_object_id_t));
+        port_attr.value.objlist.list[0] = sessionId;
+    }
+    else
+    {
+        port_attr.value.objlist.count = 0;
+    }
+
+    if (port.m_type == Port::LAG)
+    {
+        vector<Port> portv;
+        m_portsOrch->getLagMember(port, portv);
+        for (const auto p : portv)
+        {
+            status = sai_port_api->set_port_attribute(p.m_port_id, &port_attr);
+            if (status != SAI_STATUS_SUCCESS)
+            {
+                SWSS_LOG_ERROR("Failed to configure mirror session port %s: %s, status %d, sessionId %x\n",
+                                port.m_alias.c_str(), p.m_alias.c_str(), status, sessionId);
+                return false;
+            }
+        }
+    }
+    else if (port.m_type == Port::PHY)
```

Comment:
I asked this up here too - but what happens if the user specifies a PHY port that is a LAG member?

Reply:
No. LAG members cant be part of this list
I handled the cases where LAG port is also in source port list.. 

In swss:
	1. check port is part of Port channel and in src_port list.
        2. for lag update events, skip programming when the port is already part of mirror session (not possible case, but error check)

Reply:
So I see both of those checks in the code, but let's say that the src port list only has PHY ports. Do we ever check if any of those PHY ports are LAG members?

Reply:
No. we dont check that. 
Its up to user when he wants to mirror only specific ports of LAG, its allowed. 
Do u see any issue with that ?

Reply:
Ohhh, okay! I think that is OK then. Just to clarify, you can have LAGs OR LAG members, you just can't mix both in the same src port list?

Reply:
Yes. thats correct.